src/service/data/storage/pvc.py

```
@@ -610,7 +617,7 @@ async def get_partial_payload(self, payload_id: str, is_input: bool, is_modelmes
                     serialized_data = dataset.attrs[payload_id]
 
                     try:
-                        payload_dict = pkl.loads(serialized_data)
+                        payload_dict = pkl.loads(serialized_data)  # nosec B301 - Internal data from HDF5
```

Ruff (0.15.7)

[error] `pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)